tests/baselines/reference/elementAccessChain.3.types

```diff
@@ -170,7 +170,7 @@ for (obj?.a["b"] of []);
 
 [...obj?.["a"]] = [];
 >[...obj?.["a"]] = [] : never[]
->[...obj?.["a"]] : never[]
+>[...obj?.["a"]] : any[]

```

Comment:
There a reason for this change?

Reply:
It's an effect of simplifications I made to `checkArrayLiteral`. Given that `obj` is of type `any`, the type of `[...obj?.["a"]]` certainly should be `any[]`. Not sure how the old (and quite convoluted) logic got to `never[]`.